fuzz/src/chanmon_consistency.rs

```diff
@@ -546,23 +548,75 @@ pub fn do_test<Out: test_logger::Output>(data: &[u8], out: Out) {
 	}
 
 	loop {
+		// Push any events from Node B onto ba_events and bc_events
+		macro_rules! push_excess_b_events {
+			($excess_events: expr, $expect_drop_node: expr) => { {
+				let a_id = nodes[0].get_our_node_id();
+				let expect_drop_node: Option<usize> = $expect_drop_node;
+				let expect_drop_id = if let Some(id) = expect_drop_node { Some(nodes[id].get_our_node_id()) } else { None };
+				for event in $excess_events {
+					let push_a = match event {
+						events::MessageSendEvent::UpdateHTLCs { ref node_id, .. } => {
+							if Some(*node_id) == expect_drop_id { panic!("peer_disconnected should drop msgs bound for the disconnected peer"); }
+							*node_id == a_id
+						},
+						events::MessageSendEvent::SendRevokeAndACK { ref node_id, .. } => {
+							if Some(*node_id) == expect_drop_id { panic!("peer_disconnected should drop msgs bound for the disconnected peer"); }
+							*node_id == a_id
+						},
+						events::MessageSendEvent::SendChannelReestablish { ref node_id, .. } => {
+							if Some(*node_id) == expect_drop_id { panic!("peer_disconnected should drop msgs bound for the disconnected peer"); }
+							*node_id == a_id
+						},
+						events::MessageSendEvent::SendFundingLocked { .. } => continue,
+						events::MessageSendEvent::SendAnnouncementSignatures { .. } => continue,
+						events::MessageSendEvent::PaymentFailureNetworkUpdate { .. } => continue,
+						_ => panic!("Unhandled message event"),
+					};
+					if push_a { ba_events.push(event); } else { bc_events.push(event); }
+				}
+			} }
+		}
+
+		#[derive(PartialEq)]
+		enum ProcessMessages {
+			AllMessages,
+			OneMessage,
+			OnePendingMessage,
+		}
```

Comment:
After reading through the code, I understand how this is used. But I think a comment would helpful for posterity.